src/bindgen/ir/ty.rs

```diff
@@ -519,59 +519,115 @@ impl Type {
+                                        eraser.erase_transparent_types(library, ty, mappings);
+                                    if erased_ty.is_some() {
+                                        did_erase_generics = true;
+                                    };
+                                    GenericArgument::Type(erased_ty.unwrap_or_else(|| ty.clone()))
+                                }
+                                other => other.clone(),
+                            })
+                            .collect();
+
+                        // Replace transparent items with their underlying type and erase it.
+                        let aliased_ty = match item {
+                            ItemContainer::OpaqueItem(o) => o.as_transparent_alias(&generics),
+                            ItemContainer::Typedef(t) => t.as_transparent_alias(&generics),
+                            ItemContainer::Struct(s) => s.as_transparent_alias(&generics),
+                            _ => None,
+                        };
+                        if let Some(mut ty) = aliased_ty {
+                            eraser.erase_transparent_types_inplace(library, &mut ty, mappings);
+                            return Some(ty);
+                        } else if did_erase_generics {
+                            // The type was not transparent, but some of its generics were erased.
+                            let path = GenericPath::new(path.path().clone(), generics);
+                            return Some(Type::Path(path));
+                        }
+                    }
+                } else {
+                    warn!(
+                        "Can't find {}. This usually means that this type was incompatible or \
+                           not found.",
+                        path.path()
+                    );
+                }
+            }
+            Type::Primitive(..) => {} // cannot simplify further
+            Type::Array(ref ty, ref constexpr) => {
+                if let Some(erased_ty) = eraser.erase_transparent_types(library, ty, mappings) {
+                    return Some(Type::Array(Box::new(erased_ty), constexpr.clone()));
+                }
+            }
+            Type::FuncPtr {
+                ref ret,
+                ref args,
+                is_nullable,
+                never_return,
+            } => {
+                // Attempt to erase ret and all args; if any of them were actually erased, then
+                // assemble and return the simplified function signature that results.
+                let mut erased_any = false;
+                let mut try_erase = |ty| {
+                    if let Some(erased) = eraser.erase_transparent_types(library, ty, mappings) {
+                        erased_any = true;
+                        erased
+                    } else {
+                        ty.clone()
+                    }
+                };
+                let erased_ret = try_erase(ret);
```

Comment:
A bit unfortunate to do all this cloning unconditionally just to ~almost always throw it away, fwiw...